packages/amplify-graphql-conversation-transformer/src/resolvers/invoke-lambda-resolver-fn.js

```diff
@@ -0,0 +1,56 @@
+import { util } from '@aws-appsync/utils';
+
+export function request(ctx) {
+  const { args, request, prev } = ctx;
+  [[TOOL_DEFINITIONS_LINE]]
```

Comment:
Here and throughout: Does your template substitution logic guarantee that these will be:
1. Always one line?
2. Always appropriately escaped for the expected quotes?

How can we ensure that the template always uses the expected quoting scheme so the caller passing the substitution values always escapes appropriately?

Reply:
> 1. Always one line?

Not always one line. For example, `[[TOOL_DEFINITIONS_LINE]]` here can be:
```js
const dataTools = toolDefinitions.tools;
const toolsConfiguration = {
  dataTools,
  clientTools,
};
```

> 2. Always appropriately escaped for the expected quotes?

The only safeguard today are the existing snapshots, which is admittedly not great. I plan to add integration tests using AppSync's `EvaluateCode` API for the resolver functions in the future ([reference](https://docs.aws.amazon.com/appsync/latest/devguide/test-debug-resolvers-js.html)). That will give us confidence in future changes.

> How can we ensure that the template always uses the expected quoting scheme so the caller passing the substitution values always escapes appropriately?

Same answer as 2. Additional context -- the majority of these values are derived from GraphQL inputs (field name, structured directive `input` types) where values that would cause escaping issues would have failed upstream in GraphQL validation. The exceptions to this rule are the model identifier and system prompt, which are string inputs provided when defining a conversation route. The system prompt is already stringified to prevent issues here. The model identifier can also be stringified prior to substitution -- I'll look at this in a follow up.